docs/docs/features/backlight.md

```diff
@@ -0,0 +1,213 @@
+
+## Configuring Backlight
+
+There are various Kconfig options used to configure the backlight feature. These can all be set in the `.conf` file.
+
+| Option                               | Description                                           | Default |
+| ------------------------------------ | ----------------------------------------------------- | ------- |
+| `CONFIG_ZMK_BACKLIGHT_BRT_STEP`      | Brightness step in percent                            | 20      |
+| `CONFIG_ZMK_BACKLIGHT_BRT_START`     | Default brightness in percent                         | 40      |
+| `CONFIG_ZMK_BACKLIGHT_ON_START`      | Default backlight state                               | y       |
+| `CONFIG_ZMK_BACKLIGHT_AUTO_OFF_IDLE` | Turn off backlight when keyboard goes into idle state | n       |
+| `CONFIG_ZMK_BACKLIGHT_AUTO_OFF_USB`  | Turn off backlight when USB is disconnected           | n       |
+
+## Adding Backlight to a board or a shield
+
+<Tabs
+defaultValue="shield"
+values={[
+{label: 'Adding to a board', value: 'board'},
+{label: 'Adding to a shield', value: 'shield'},
+]}>
+<TabItem value="board">
+
+First, you must enable PWM by adding the following lines to your `.conf` file:
+
+```
+CONFIG_PWM=y
+CONFIG_LED_PWM=y
+```
+
+Then you have to add the following lines to your `.dts` file:
+
+```
+&pwm0 {
+	status = "okay";
+	ch0-pin = <45>;
+	/* ch0-inverted; */
+};
+```
+
+The value `ch0-pin` represents the pin that controls the LEDs. With nRF52 boards, you can calculate the value to use in the following way: you need the hardware port and run it through a function.
+**32 \* X + Y** = `<Pin number>` where X is first part of the hardware port "PX.01" and Y is the second part of the hardware port "P1.Y".
+
+For example, _P1.13_ would give you _32 \* 1 + 13_ = `<45>` and _P0.15_ would give you _32 \* 0 + 15_ = `<15>`.
+
+If your board uses a P-channel MOSFET to control backlight instead of a N-channel MOSFET, you may want to enable `ch0-inverted`.
+
+Then you have to add the following lines inside the root devicetree node on the same file as before:
+
+```
+/ {
+    backlight: pwmleds {
+        compatible = "pwm-leds";
+        label = "Backlight LEDs";
+        pwm_led_0 {
+            pwms = <&pwm0 45>;
+        	label = "Backlight LED 0";
+        };
+    };
+};
+```
+
+The value inside `pwm_led_0` must be the same as you used before.
+
+:::info
+Note that every LED inside of the backlight node will be treated as a backlight LED, so if you have other PWM LEDs you need to declare them in a separate node. Refer to [Multiple backlight LEDs](#multiple-backlight-leds) if you have multiple backlight LEDs.
+:::
+
+Finally you need to add backlight to the `chosen` element of the root devicetree node:
+
+```
+/ {
+    chosen {
+        zmk,backlight = &backlight;
+    };
+}:
+```
+
+</TabItem>
+<TabItem value="shield">
+
+First, you must enable PWM by adding the following lines to your `.conf` file:
```

Comment:
This would usually go in the shield's `boards/<board>.conf` file.

Reply:
Fixed